### PR TITLE
Bug fix for __read_csv() method to now read RISMA network data of "header & files" format again

### PR DESCRIPTION
**Pull request by nfb2021**
General:
---------
There was a Problem with reading data from the RISMA network. 
This data only contained soil moisture as variable and was of format  "header & files".
Further, this data was dowloaded from the ISMN webpage May 22, 23 at 09:00.

Problem:
----------
To my understanding, all sensor files of the "header & files" format should contain four columns: date/time, data, new flag, old flag.
For some RISMA sensor files the last column, old flag, contained strings such as:
    - "Frozen soil,OK,OK,OK,OK,OK" (see **RISMA_RISMA_MB11_sm_1.000000_1.000000_Hydraprobe-II-Sdi-12-A_19500101_20230521.stm**)
    - "DLT >= 1.5,OK,OK,OK,OK,OK" (see **RISMA_RISMA_MB11_sm_0.000000_0.050000_Hydraprobe-II-Sdi-12-C_19500101_20230521.stm**)
    - "Out of WFV average range,NA,NA,OK,OK,OK" (see **RISMA_RISMA_CEF_sm_0.000000_0.050000_Hydraprobe-II-Sdi-12-B_19500101_20230521.stm**) 

The method ``__read_csv()`` of the ``DataFile`` class could not parse these sensor files and threw following error: **pandas.errors.ParserError: Error tokenizing data. C error: Expected 5 fields in line 23, saw 9**.
This phenomenon only occurred with some RISMA sensor files, other networks were fine.

Fix:
------
- Extended the ``__read_csv()`` method to take keyword arguments ``**kwargs`` as input
- Replaced lambda function ``readf()`` calling the ``pandas.read_csv()`` with a method of same name, that takes input arguments
- Defined additional default arguments: ``delim_whitespace=None, sep=None, low_memory=None``
- With a try and except block caught ParserError, and then call an adapted ``pd.read_csv()``, to fit the problems mentioned above
- Subsequently adapted the ``__read_format_header_values()`` method correctly to call the ``__read_csv()`` in its ``return`` statement

Note:
-----
- unit tests ran successfully
- data of format "CEOP" was NOT tested and the bug fix does NOT apply to it

### CODE REVIEW

File: .vscode/settings.json
Comment (wpreimes) [inline .vscode/settings.json:1]: Can you put `.vscode/*` into the .gitignore file in order to not include the editor settings?

File: changes_done/changes_done.rst
Comment (wpreimes) [inline changes_done/changes_done.rst:1]: You can delte this and add it ass a comment to the Pull Request

File: filelist.txt
Comment (wpreimes) [inline filelist.txt:1]: can this be deleted?

File: fl_elements.txt
Comment (wpreimes) [inline fl_elements.txt:1]: can this be deleted?

File: src/ismn/components.py [hunk 1]
```diff
@@ -467,10 +467,14 @@ def get_sensors(self, variable, depth_from, depth_to):
         sensors : numpy.ndarray
             array of sensors found for the given combination of variable and depths
         """
-        return np.array([
-            s for s in self.iter_sensors(
-                variable=variable, depth=Depth(depth_from, depth_to))
-        ])
+        return np.array(
```
Comment (wpreimes) [inline src/ismn/components.py:470]: this looks like black, can you apply `yapf file.py --in-place` to all files where you changed anything?

File: src/ismn/custom.py [hunk 2]
```diff
@@ -82,21 +82,21 @@ class CustomStationMetadataCsv(CustomMetaReader):
...
-        ----------
-        station_meta_csv: str
-            Path to the csv file with the above described content
-       fill_values: dict, optional (default: None)
-            Values to use for a certain custom metadata variable, if no
-            match is found.
-        kwargs:
-            Additional kwargs as passed to :func:`pandas.read_csv`
-            To use a different separator than the default semicolon, use `sep`
+         Parameters
```
Comment (wpreimes) [inline src/ismn/custom.py:85]: This looks not intentional

File: src/ismn/filehandlers.py [hunk 3]
```diff
@@ -22,6 +22,9 @@
 
 import os
 import pandas as pd
+import warnings
+
+warnings.simplefilter(action="ignore", category=UserWarning)
```
Comment (wpreimes) [inline src/ismn/filehandlers.py:27]: This catches all warning no? Is this intentional? generally warning are useful

**Review by wpreimes — CHANGES_REQUESTED** [review]

**Comment by wpreimes** [comment]
I see that the pipeline for windows tests is not working in master right now. I will fix this. Afterwards you update your branch with the changes in master and then the tests should run here as well.

**Comment by wpreimes** [comment]
Regarding test data, it's fine to add a new file to check the new changes. Generally we should keep the number of test files low, so maybe if we don't need all of the newly added files, you can delete some of them?

File: src/ismn/custom.py (hunk 2 above)
Comment (nfb2021) [reply src/ismn/custom.py:85]: I don't think I did that. Just checked the TUW-GEO/ismn repo and it already contains this piece of code, last committed 4 months ago. If I did it, then completely unintentionally

File: src/ismn/filehandlers.py (hunk 3 above)
Comment (nfb2021) [reply src/ismn/filehandlers.py:27]: Was a leftover from debugging, removed it now

File: src/ismn/custom.py (hunk 2 above)
Comment (wpreimes) [reply src/ismn/custom.py:85]: ah true

**Comment by wpreimes** [comment]
and please add one sentence on the fixed bug to CHANGELOG.rst (under unreleased changes)

**Comment by wpreimes** [comment]
ok, looks good. Tests are passing now. Do you still want to change anything here?

**Comment by nfb2021** [comment]
> I just noticed two more things :)
> 
>     1. the folder .vscode is still there, I assume because you committed before updating the .gitignore file. There is a command to delete an (ignore) folder from the git history. Can you check that? I want to avoid having that in the upstream repo

It should be removed now from git

>     2. The folder RISMA_test_data should either be deleted if it's not needed anymore or integrated into the actual test data if you want to include reading the data as part of the available tests. Both are fine for me. If you want to include the data, please copy the network folder to  e.g. test/test_data/Data_seperate_files_header_20170810_20180809, make sure in the tests, that the RISMA data is correctly read, and if necessary update the tests.

I just deleted the directory. With everything going on with the pull request that is already quite some input. Adapting the tests is something for the next time

**Comment by wpreimes** [comment]
great, thanks.

**Pull request merged**